src/fights.ts

```diff
@@ -1238,9 +1238,16 @@ const freeFightSources = [
   new FreeFight(
     () => get("snojoAvailable") && clamp(10 - get("_snojoFreeFights"), 0, 10),
     () => {
-      if (get("snojoSetting") === null) {
+      const snojoSetting = get("snojoSetting", "NONE");
+      if (snojoSetting === "MYSTICALITY" && get("snojoMysticalityWins") > 50) {
         visitUrl("place.php?whichplace=snojo&action=snojo_controller");
         runChoice(3);
+      } else if (snojoSetting === "MUSCLE" && get("snojoMuscleWins") > 50) {
+        visitUrl("place.php?whichplace=snojo&action=snojo_controller");
+        runChoice(2);
+      } else if (snojoSetting === "NONE") {
+        visitUrl("place.php?whichplace=snojo&action=snojo_controller");
+        runChoice(1);
```

Comment:
We should make setting the snojo up a daily task instead of being part of this FreeFight, probably.

Then in the task we can have something like ```
type SnojoSetting = "MUSCLE" | "MYSTICALITY" | "MOXIE" | "NONE";

const SETTINGS = {
  MOXIE: { choice: 1, pref: "snojoMoxieWins", },
  MUSCLE: { choice: 2, pref: "snojoMuscleWins", },
  MYSTICALITY: { choice: 3, pref: "snojoMysticalityWins", },
} as const;

const currentSetting = get<SnojoSetting>("snojoSetting", "NONE");
```
and go from there.

But I think now that this logic is becoming something other than "always pick x", it needs to move to dailies.

Reply:
I guess if it's being split out I should do it properly.

The calculation is something like:
* if you're ascending pick the highest of the values of each of the consumables
* otherwise assume you're in for at least 5 days, and calculate the higher of (scroll + 7 * consumables) / 5 or each consumable and pick accordingly, discarding options where you've already got the scroll.